lib/explorer/series.ex

```diff
@@ -3167,10 +3171,20 @@ defmodule Explorer.Series do
     end
   end
 
-  # TODO: fix the logic for integer dtypes
-  defp cast_to_add({:s, left}, {:s, right}), do: {:s, max(left, right)}
-  defp cast_to_add({:s, _}, {:f, _} = float), do: float
-  defp cast_to_add({:f, _} = float, {:s, _}), do: float
+  # Review the size needed for this operation.
+  defp cast_to_add({int_type, left}, {int_type, right}) when K.in(int_type, [:s, :u]),
+    do: {int_type, max(left, right)}
+
+  defp cast_to_add({:s, s_size}, {:u, u_size}) when u_size >= s_size,
+    do: {:s, min(64, u_size * 2)}
+
+  defp cast_to_add({:u, u_size}, {:s, s_size}) when u_size >= s_size,
+    do: {:s, min(64, u_size * 2)}
+
+  defp cast_to_add({:s, s_size}, {:u, _}), do: {:s, s_size}
+  defp cast_to_add({:u, _}, {:s, s_size}), do: {:s, s_size}
```

Comment:
```suggestion
  defp cast_to_add({:s, s_size}, {:u, u_size}), do: min(64, max(s_size, 2 * u_size))
  defp cast_to_add({:u, s_size}, {:s, u_size}), do: min(64, max(s_size, 2 * u_size))
```

I think this works here (or wherever this logic ends up).

Reply:
@billylanchantin makes sense! Thanks!